fred2/campaigntreewnd.cpp

```diff
@@ -133,8 +132,9 @@ void campaign_tree_wnd::OnCpgnFileOpen()
 		if (!strlen(name))
 			return;
 
-		string_copy(Campaign.filename, name, MAX_FILENAME_LEN - 1);
-		Campaign_tree_formp->load_campaign();
+		full_path = dlg.GetPathName();
```

Comment:
How long is that pointer valid? I don't really see anything in the documentation, but this might rely on it being valid and unchanged longer than is safe to do so.

Reply:
Good point.  I changed things up a bit.